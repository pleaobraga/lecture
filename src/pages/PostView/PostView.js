import React, { Component } from 'react'
import PostDetail from '../../components/Post/PostDetail/PostDetail'
import { connect } from 'react-redux'
import * as actions from '../../actions/index'


export class PostView extends Component {

    componentDidMount() {
        const { idPost } = this.props.match.params
        this.props.getPostDetail(idPost)
    }

    render() {
        const { post } = this.props

        return (
            <PostDetail post={post} />
        );
    }
} 


const mapStateToProps = state => ({
    post: state.post.postDetail
})

const mapDispatchToProps = dispatch => ({
    getPostDetail: (idPost) => dispatch(actions.post.getPostDetail(idPost))
})


export default connect(mapStateToProps, mapDispatchToProps)(PostView)