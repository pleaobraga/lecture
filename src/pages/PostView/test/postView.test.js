import React from 'react'
import { shallow } from 'enzyme'
import { PostView } from '../PostView'

describe('PostView', () => {
    
    let postView = shallow(<PostView />)


    it('render Proprely', () => {
        expect(postView).toMatchSnapshot()
    })

    it('contains a connected Post Detail coponent', () => {
        expect(postView.find('PostDetail').exists()).toBe(true);
    })
})