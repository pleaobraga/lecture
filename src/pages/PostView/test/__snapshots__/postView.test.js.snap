// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PostView render Proprely 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <PostView />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "function",
    "props": Object {
      "post": undefined,
    },
    "ref": null,
    "rendered": null,
    "type": [Function],
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "function",
      "props": Object {
        "post": undefined,
      },
      "ref": null,
      "rendered": null,
      "type": [Function],
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
    "disableLifecycleMethods": true,
  },
}
`;
